torch/csrc/jit/passes/freeze_module.cpp

```diff
@@ -14,7 +14,33 @@ namespace {
 
 class AttributePropagator {
  public:
-  AttributePropagator(Module& module) : module_(module) {}
+  AttributePropagator(Module& module, std::vector<std::string>& preservedAttrs)
+      : module_(module) {
+    // Currently only top level attributes and functions can  be preserved
+    // explicitely.
+    auto checkName = [this](std::string& name) {
+      if (module_.hasattr(name)) {
+        insertMutableAttr(name, module_.attr(name), module_._ivalue());
+        return true;
+      }
+
+      for (auto& fn : module_.type()->methods()) {
+        if (fn->name() == name) {
+          preservedMethods_.insert(fn);
+          return true;
+        }
+      }
+      return false;
+    };
+
+    // forward is preserved by default.
```

Comment:
I don't know if we want this - if this is true there will be no way to remove the forward function. I think the signature should just be `freeze_module(module, List[str])` and then `freeze_module(module)` which will freeze all but forward.